packages/peregrine/lib/talons/CheckoutPage/ShippingInformation/useShippingInformation.js

```diff
@@ -0,0 +1,79 @@
+import { useCallback, useEffect, useMemo } from 'react';
+import { useLazyQuery } from '@apollo/react-hooks';
+
+import { useAppContext } from '../../../context/app';
+import { useCartContext } from '../../../context/cart';
+import { MOCKED_ADDRESS } from '../../CartPage/PriceAdjustments/ShippingMethods/useShippingForm';
+
+export const useShippingInformation = props => {
+    const {
+        onSave,
+        queries: { getShippingInformationQuery }
+    } = props;
+
+    const [, { toggleDrawer }] = useAppContext();
+    const [{ cartId }] = useCartContext();
+
+    const [fetchShippingInformation, { called, data, loading }] = useLazyQuery(
+        getShippingInformationQuery
+    );
+
+    useEffect(() => {
+        if (cartId) {
+            fetchShippingInformation({
+                variables: {
+                    cartId
+                }
+            });
+        }
+    }, [cartId, fetchShippingInformation]);
+
+    const shippingData = useMemo(() => {
+        let filteredData;
+        if (data) {
+            const { cart } = data;
+            const { email, shipping_addresses: shippingAddresses } = cart;
+            if (shippingAddresses.length) {
+                const primaryAddress = shippingAddresses[0];
+                for (const field in MOCKED_ADDRESS) {
+                    if (primaryAddress[field] === MOCKED_ADDRESS[field]) {
+                        primaryAddress[field] = '';
+                    }
+
+                    if (
+                        field === 'street' &&
+                        primaryAddress[field][0] === MOCKED_ADDRESS[field][0]
+                    ) {
+                        primaryAddress[field] = [''];
+                    }
+                }
+
+                filteredData = {
+                    email,
+                    ...primaryAddress
+                };
+            }
+        }
+
+        return filteredData;
+    }, [data]);
+
+    const doneEditing = shippingData && shippingData.email;
```

Comment:
This is potentially flaky - while we don't require an email in cart right now it's possible we may add it later which would cause this form to render the card view. Also considering auth checkout that may have an email pre-attached to the cart but not other info. I know there is probably a follow up ticket for auth but I wanted to call this out here.

Reply:
This duck typing seems fine in this scope. I agree it may need to change if there are edge cases introduced by the auth flow or other interactions in the app, and should be addressed in that scope.

Reply:
I just saw your comment on Andy's PR, and see the need to standardize this now. This might be a good candidate for another `@client` field on the cart, so each component doesn't have to guess. Let's discuss more during our peer review tomorrow.